core/spec/models/spree/order_contents_spec.rb

```diff
@@ -159,6 +172,24 @@
       expect(line_item.quantity).to eq(2)
     end
 
+    context 'when store_credits payment' do
+      let(:payment) { create(:store_credit_payment, order: order) }
+
+      before do
+        subject.add(variant, 1)
+        payment
+      end
+
+      it 'is expected to have store_credit_payment' do
+        expect(order.payments.store_credits).to include(payment)
+      end
+
+      it "is expected to delete store_credits payment" do
```

Comment:
It would be nicer to use RSpec [change matcher](https://www.relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/change-matcher) :)